lib/oli_web/components/delivery/practice_activities/practice_activities.ex

```diff
@@ -0,0 +1,1012 @@
+        end
+      end)
+
+    activity
+    |> Map.put(:students_with_attempts_count, Enum.count(students_with_attempts))
+    |> Map.put(:student_emails_without_attempts, student_emails_without_attempts)
+    |> Map.put(:total_attempts_count, count_attempts(activity, section, student_ids) || 0)
+  end
+
+  defp get_preview_rendered(activity, socket) do
+    case get_activity_details(
+           activity,
+           socket.assigns.section,
+           socket.assigns.activity_types_map
+         ) do
+      nil ->
+        socket
+
+      activity_attempt ->
+        part_attempts = Core.get_latest_part_attempts(activity_attempt.attempt_guid)
+
+        rendering_context =
+          OliWeb.ManualGrading.Rendering.create_rendering_context(
+            activity_attempt,
+            part_attempts,
+            socket.assigns.activity_types_map,
+            socket.assigns.section
+          )
+          |> Map.merge(%{is_liveview: true})
+
+        OliWeb.ManualGrading.Rendering.render(
+          rendering_context,
+          :instructor_preview
+        )
+    end
+  end
+
+  defp assign_selected_assessment(socket, selected_assessment_id)
+       when selected_assessment_id in ["", nil] do
+    case socket.assigns.table_model.rows do
+      [] ->
+        socket
+
+      rows ->
+        assign_selected_assessment(socket, hd(rows).resource_id)
+    end
+  end
+
+  defp assign_selected_assessment(socket, selected_assessment_id) do
+    table_model =
+      Map.merge(socket.assigns.table_model, %{
+        selected: "#{selected_assessment_id}"
+      })
+
+    assign(socket, table_model: table_model)
+  end
+
+  defp count_attempts(
+         current_activity,
+         %Section{analytics_version: :v2, id: section_id},
+         student_ids
+       ) do
+    page_type_id = Oli.Resources.ResourceType.get_id_by_type("activity")
+
+    from(rs in ResourceSummary,
+      where:
+        rs.section_id == ^section_id and rs.resource_id == ^current_activity.resource_id and
+          rs.user_id in ^student_ids and rs.project_id == -1 and rs.publication_id == -1 and
+          rs.resource_type_id == ^page_type_id,
+      select: sum(rs.num_attempts)
+    )
+    |> Repo.one()
+  end
+
+  defp count_attempts(current_activity, section, student_ids) do
+    from(ra in ResourceAttempt,
+      join: access in ResourceAccess,
+      on: access.id == ra.resource_access_id,
+      where:
+        ra.lifecycle_state == :evaluated and access.section_id == ^section.id and
+          access.resource_id == ^current_activity.resource_id and access.user_id in ^student_ids,
+      select: count(ra.id)
+    )
+    |> Repo.one()
+  end
+
+  defp get_activities(current_assessment, section, student_ids) do
+    activities =
+      from(aa in ActivityAttempt,
+        join: res_attempt in ResourceAttempt,
+        on: aa.resource_attempt_id == res_attempt.id,
+        where: res_attempt.lifecycle_state == :evaluated and aa.lifecycle_state == :evaluated,
```

Comment:
The `res_attempt.lifecycle_state == :evaluated` clause must be removed.  Practice pages will usually never have their lifecycle_state transitioned into `:evaluated`